examples/immediate_multiple_requests.rs

```diff
@@ -0,0 +1,151 @@
+
+    let x: i32 = 100;
+    let mut y: i32 = 0;
+
+    mpi::request::scope(|scope| {
+        let sreq = world.this_process().immediate_send(scope, &x);
+        let rreq = world.any_process().immediate_receive_into(scope, &mut y);
+        let result = rreq.wait_for_data();
+        assert_eq!(*result, x);
+        sreq.wait();
+    });
+
+    // Test wait_any()
+    let mut result: Vec<i32> = vec![0; COUNT];
+    mpi::request::multiple_scope(2 * COUNT, |scope, coll| {
+        let prev_proc = if rank > 0 { rank - 1 } else { size - 1 };
+        let next_proc = (rank + 1) % size;
+        for _ in 0..result.len() {
+            let sreq = world
+                .process_at_rank(next_proc)
+                .immediate_send(scope, &rank);
+            coll.add(sreq);
+        }
+        for val in result.iter_mut() {
+            let rreq = world
+                .process_at_rank(prev_proc)
+                .immediate_receive_into(scope, val);
+            coll.add(rreq);
+        }
+        let mut send_count = 0;
+        let mut recv_count = 0;
+        while coll.incomplete() > 0 {
+            let (_, _, result) = coll.wait_any().unwrap();
+            if *result == rank {
+                send_count += 1;
+            } else {
+                recv_count += 1;
+            }
+        }
+        assert_eq!(send_count, COUNT);
+        assert_eq!(recv_count, COUNT);
+    });
+
+    let mut result: Vec<i32> = vec![0; COUNT];
+    // Test wait_some()
+    mpi::request::multiple_scope(2 * COUNT, |scope, coll| {
+        let prev_proc = if rank > 0 { rank - 1 } else { size - 1 };
+        let next_proc = (rank + 1) % size;
+        for _ in 0..result.len() {
+            let sreq = world
+                .process_at_rank(next_proc)
+                .immediate_send(scope, &rank);
+            coll.add(sreq);
+        }
+        for val in result.iter_mut() {
+            let rreq = world
+                .process_at_rank(prev_proc)
+                .immediate_receive_into(scope, val);
+            coll.add(rreq);
+        }
+        let mut send_count = 0;
+        let mut recv_count = 0;
+        let mut some_buf = vec![];
+        let mut finished = 0;
+        while coll.incomplete() > 0 {
+            coll.wait_some(&mut some_buf);
+            println!("wait_some(): {} request(s) completed", some_buf.len());
+            finished += some_buf.len();
+            assert_eq!(coll.incomplete(), 2 * COUNT - finished);
+            for (_, _, result) in some_buf.iter() {
+                if **result == rank {
+                    send_count += 1;
+                } else {
+                    recv_count += 1;
+                }
+            }
+        }
+        assert_eq!(send_count, COUNT);
+        assert_eq!(recv_count, COUNT);
+    });
+
+    let mut result: Vec<i32> = vec![0; COUNT];
+    // Test wait_all()
+    mpi::request::multiple_scope(2 * COUNT, |scope, coll| {
+        let prev_proc = if rank > 0 { rank - 1 } else { size - 1 };
```

Comment:
This is the standard pattern in MPI codes, but I see rsmpi uses the one you have in other places so I can update them all later.
```suggestion
        let prev_proc = (rank - 1 + size) % size;
```